### DISCUSSION TITLE
Is it possible to run this with 8 GB memory on Mac?

### DISCUSSION THREAD

**Discussion by Fiwx** [PGScatalog/pgsc_calc#147] [Q&A] [answered]
Hi, I am on a Mac with 8 GB memory. I was wondering if it is possible to run pgsc_calc, and if so, what parameters to modify to allow this. Thank you.

**Comment by nebfield** [comment, marked as answer]
You could try 😅 It depends on the size and complexity of your target genomes and scoring files. 

First you'd need to increase the [amount of memory / CPU in Docker Desktop](https://docs.docker.com/desktop/settings/mac/). The default limits are quite low.

Then just follow the instructions [here to change pipeline resources](https://pgsc-calc.readthedocs.io/en/latest/how-to/bigjob.html#configuring-pgsc-calc-to-use-more-resources-locally) (although instead of using more resources, you want to use less). As a quick example, make a new text file that contains:

```
process {
    executor = 'local'

    withLabel:process_low {
        cpus   = 1
        memory = 8.GB
        time   = 1.h
    }
    withLabel:process_medium {
        cpus   = 2
        memory = 8.GB
        time   = 4.h
    }
    withLabel:process_high_memory {
        cpus   = 2
        memory = 8.GB
        time   = 4.h
    }
}
```

and save it as something like `custom.config`

then run:

```
$ nextflow run pgscatalog/pgsc_calc -c path/to/custom.config ... (your normal parameters)
```

When you run the pipeline, if you get any errors with `exit code 137` that means you've run out of memory and the workflow was automatically killed by docker 😬 There's no fix for this, so you'd need to find a computer with more memory.

**Reply by smlmbrt** [reply]
It also depends on how many variants are in your genotyping data (the fewer the better in this case) - plink will error out if it doesn't get >8 Gb for the LD-pruning step.
